### PR TITLE
OBPIH-3365 Add search to dashboard filter

### PR DESCRIPTION
**Pull request by julienreliefapps**

### CODE REVIEW

File: src/js/components/tablero/Filter.jsx [hunk 1]
```diff
@@ -39,6 +53,8 @@ class Filter extends Component {
     if (categoryData) this.getCategoryRows(categoryData.endpoint);
     this.setState({ titlePopup: nameCategory || 'Add Filter' });
     this.setState({ categorySelected: nameCategory || '' });
+    // Value in the searchBar removed
+    document.getElementById('searchBar').value = '';
```
Comment (awalkowiak) [inline src/js/components/tablero/Filter.jsx:57]: Add a searchTerm to the state and set it here to empty value.

File: src/js/components/tablero/Filter.jsx [hunk 2]
```diff
@@ -163,6 +179,13 @@ class Filter extends Component {
                 > X
                 </span>
               </div>
+              <input
```
Comment (awalkowiak) [inline src/js/components/tablero/Filter.jsx:182]: Add value property in this input: `value={this.state.searchTerm}`

File: src/js/components/tablero/Filter.jsx [hunk 3]
```diff
@@ -25,9 +25,23 @@ class Filter extends Component {
...
+        listCategoryDataFiltered: [],
+      }));
+  }
+
+  searchOnChange = (event) => {
+    const filteredList = this.state.listCategoryData
+      .filter(categoryData => categoryData.name
+        .toLowerCase()
+        .includes(event.target.value));
+    this.setState({ listCategoryDataFiltered: filteredList });
```
Comment (awalkowiak) [inline src/js/components/tablero/Filter.jsx:44]: Set `searchTerm: event.target.value` state property

File: src/js/components/tablero/tablero.scss [hunk 4]
```diff
@@ -72,12 +74,26 @@
             cursor: pointer;
           }
         }
+        input {
```
Comment (awalkowiak) [inline src/js/components/tablero/tablero.scss:77]: When you click on the input field within the filter, then the black border does not have rounded corners.

**Review by awalkowiak — CHANGES_REQUESTED** [review]
This filtering does not work for me locally, please fix it (I shared the gif explaining it in the JIRA ticket)

File: src/js/components/tablero/Filter.jsx (hunk 1 above)
Comment (julienreliefapps) [reply src/js/components/tablero/Filter.jsx]: done

File: src/js/components/tablero/Filter.jsx (hunk 2 above)
Comment (julienreliefapps) [reply src/js/components/tablero/Filter.jsx:186]: Done

File: src/js/components/tablero/Filter.jsx (hunk 3 above)
Comment (julienreliefapps) [reply src/js/components/tablero/Filter.jsx]: Done

File: src/js/components/tablero/tablero.scss (hunk 4 above)
Comment (julienreliefapps) [reply src/js/components/tablero/tablero.scss:77]: Done

**Review by awalkowiak — APPROVED** [review]

**Pull request merged**